lib/index.js

```diff
@@ -0,0 +1,266 @@
+/*!
+ * Copyright (c) 2019 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+const assert = require('assert-plus');
+const base64url = require('base64url-universal');
+const bedrock = require('bedrock');
+const brPackageManager = require('bedrock-package-manager');
+const crypto = require('crypto');
+const database = require('bedrock-mongodb');
+const {promisify} = require('util');
+const {util: {BedrockError, uuid}} = bedrock;
+
+// load config defaults
+require('./config');
+
+const AES_KW_ALGORITHM = 'id-aes256-wrap';
+const AES_KW_RFC3394_IV = Buffer.from('A6A6A6A6A6A6A6A6', 'hex');
+
+bedrock.events.on('bedrock.init', () => {
+  brPackageManager.register({
+    // FIXME: is this the alias for this package? This is what is already in
```

Comment:
@dlongley need feedback here and on L27 below.

Reply:
What is "alias" used for? Maybe we want a KMS plugin name instead? Something specific to KMS?

Reply:
`alias` is just a shortname to refer to the plugin, any string will do.  Having/using an alias allows for not having to use 'bedrock-long-name' everywhere, and also provides for replacing the NPM plugin package associated with a given alias in the future.

Reply:
Ok, so that's fine. I suspect that we can't use `bedrock-ssm` here for this particular package though because other storage engines would want to fight over it. So no alias then (would be my guess).

Reply:
so you want to potentially encode this full package name into a key id as we've discussed elsewhere?  Potentially: `urn:bedrock-kms:bedrock-ssm-mongodb:b927c81f-9df4-484e-ace0-1b69db04f090`

Using an alias does allow us to be a bit more flexible.  So for instance, let's say we used an alias something like `ssmv1`.  We could potentially migrate all the keys from this module to an HSM and replace the package associated with the `ssmv1` alias to a plugin package that interfaces with that HSM.

Reply:
I really do like `ssmv1` for that reason ... but something feels off there, like the namespacing is leaking between package names and KMS plugin names. I wonder if we should be able to find and load packages based on some other property. I don't think "alias" is that property because it seems like it's at the wrong level of abstraction. Thinking...

Reply:
To add more grist to the mill...  That we do have domain specific `meta` that goes along with packages in bedrock-package-manager.  One thought I had was that we could have `meta.regexFoo` would could be a key ID matcher.  The idea was that it would be lighter than calling a method on the plugin API itself, although we *could* do that.

Reply:
If we have an alias that is the same with KMS plugins and say, ledger consensus plugins (or whatever else) ... that shouldn't cause a problem, but it would with this design. I think we should have a way to load packages from other meta properties and use that instead? For example, using `kmsPlugin: 'ssmv1'` in the `meta` for this package would avoid that problem. Then we'd load a package of type `bedrock-kms-store` with `meta.kmsPlugin === 'ssmv1'`.

Reply:
Or maybe aliases could just be type scoped.

Reply:
Yes, with the `package.type` ;)  that's already implemented, are we done here? :)

Reply:
If we can type-scope aliases I'm +1 for this feature.

> so you want to potentially encode this full package name into a key id as we've discussed elsewhere? Potentially: urn:bedrock-kms:bedrock-ssm-mongodb:b927c81f-9df4-484e-ace0-1b69db04f090

Yes, we want to do this. And, yes, it would be nicer to use an alias in there.